src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -15,11 +15,43 @@ internal class MsQuicApi : IDisposable
 
         private readonly IntPtr _registrationContext;
 
+        public static bool IsQuicSupported { get; } = GetIsQuicSupported();
+
+        private static bool GetIsQuicSupported()
+        {
+            // TODO: check using MsQuicOpen return value; but there doesn't seem to be a consistent error code for this yet.
+            OperatingSystem ver = Environment.OSVersion;
+
+            if (ver.Platform != PlatformID.Win32NT || ver.Version >= new Version(10, 0, 19041, 0))
```

Comment:
I don't think we should be checking this version of Windows here; technically, this OS version is for schannel support of TLS 1.3. There are other TLS implementation which would work on a lower windows version (I haven't tried but there are options for msquic).

Reply:
I'd like to get rid of that version check all together. It is not something that `MsQuicOpen` currently checks, and will instead result in callbacks failing with unspecified platform-specific TLS errors.

The TODO is left to find a better way. It is unclear what exactly msquic requires from kernel to be successful. The recommendation I've got so far is that "latest is better". This version check will at least avoid loading MsQuic on versions of Windows we haven't tested it against.

My primary goal is to avoid needing to comment out source code to avoid breaking CI; that's fragile and we will have an embarrassing merge at some point.